README.md
# PHP-EasyDirectoryExplorer
Easy Directory Explorer for PHP file systems


changelog.txt
Latest Versions (v1.08): <br />
&nbsp;&nbsp;Installer - v1.06 <br />
&nbsp;&nbsp;Directory Explorer - v1.06 <br />
&nbsp;&nbsp;EDEupdatehelper - v1.04 <br />
 <br /> <br />
Changelog v1.08: <br />
&nbsp;&nbsp;Installer v1.06: <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Update Installer now also updates the update helper file. <br />
&nbsp;&nbsp;EDEupdatehelper v1.04 <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Now retrieves source files from new directory. <br />
&nbsp;&nbsp;
 <br /> <br />
Changelog v1.07: <br />
&nbsp;&nbsp;Installer v1.05: <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Changes to the ways the buttons work. <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Now tells user if the key is wrong. <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Security flaw fixed. <br />
&nbsp;&nbsp;Directory Explorer v1.06 <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Fixed the sorting error, links weren't pointing the right way. <br />
&nbsp;&nbsp;EDEupdatehelper v1.03 <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Now preserves the key during the update. <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Security flaw fixed. <br />
&nbsp;&nbsp;
 <br /> <br />
Changelog v1.06: <br />
&nbsp;&nbsp;Installer v1.04: <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Added a new function that enables you to update your installer to the latest version from the hosting server. <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Added two new buttons to quickly navigate to index.php and dir.php <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Added a required checkbox to make sure you agree to the terms that were also added during this update. <br />
&nbsp;&nbsp;&nbsp;&nbsp;- Added a changelog viewer so you can check if you're up to date, and what was changed or added. <br />
&nbsp;&nbsp;Directory Explorer v1.05 <br />
&nbsp;&nbsp;&nbsp;&nbsp;- You can now sort the files by extension, sub-sorting by other parameters may be added later, but don't hold your breath. <br />
&nbsp;&nbsp;&nbsp;&nbsp;- If you add ?t=c to the end of the URL you can load the page with the subdirectory lists closed. Useful for those longer pages. <br />
&nbsp;&nbsp;EDEupdatehelper v1.01 <br />
&nbsp;&nbsp;&nbsp;&nbsp;- EDEupdatehelper was added. It will likely remain unchanged for now. <br />
&nbsp;&nbsp;

dir.php
<!--
PHP Easy File Explorer
Author: Caspar Neervoort
Creative Commons Attribution-NoDerivatives 4.0 International License.

REQUIRES: PHP 5.3 or higher.
You may use this anywhere and in any way you want, as long as you don`t alter the license information.
If you want to adapt this script, please contact me for permission and further information.
-->

<html>
<head>
	<title>Easy Directory Explorer</title>
	<META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
	<style>
		body
		{
			margin: 0;
		}
		h1
		{
			font-family: Arial, Verdana, sans-serif;
		}
		p
		{
			font-family: Arial, Verdana, sans-serif;
			font-size: 12px;
			margin-top: 10px;
			margin-bottom: 12px;
		}
		#header
		{
			z-index: 500;
			width: 100%; 
			padding-left: 5px; 
			padding-top: 5px; 
		}
		#menu
		{
			z-index: 500;
			width: 100%; 
			padding: 1px; 
			background: #adadad;
		}
		.menuSort
		{
			margin-top: -27px; 
			margin-right: 20px; 
			text-align: right;
		}
		#fixPage
		{
			z-index: 500;
			width: 100%; 
			padding: 1px; 
			visibility: hidden;
			display: none;
		}
		.dir
		{
			width: 100%; 
			padding: 1px; 
			background: #dedede;
		}
		.dir:hover
		{
			background: #cdcdcd;
		}
		.file
		{
			width: 100%; 
			padding: 1px; 
			background: #efefef;
		}
		.file:hover
		{
			background: #cdcdcd;
		}
		.toggleDir
		{
			margin-top: -27px; 
			margin-right: 20px; 
			text-align: right;
			cursor: pointer;
		}
		.anchorDir
		{
			margin-top: -27px; 
			margin-right: 45px; 
			text-align: right;
			cursor: pointer;
		}
		.fileDetails
		{
			margin-top: -27px; 
			margin-right: 20px; 
			text-align: right;
			color: #aaa;
			pointer-events: none;
		}
		.footerText
		{
			color: #999;
		}
	</style>
	<script>
		function toggleDir(dirID)
		{
			console.log(dirID);
			dirDiv = document.getElementById(dirID);
			dirToggle = document.getElementById("toggle" + dirID);
			if (dirDiv.style.display == "block")
			{
				dirDiv.style.display = "none";
				dirToggle.innerHTML = "<";
			} else
			{
				dirDiv.style.display = "block";
				dirToggle.innerHTML = "V";
			}
		}
		
		window.onscroll = function (event) 
		{	
			if(document.body.scrollTop > document.getElementById('header').offsetHeight)
			{
				document.getElementById('menu').style.position = "fixed";
				document.getElementById('menu').style.top = "0";
				document.getElementById('fixPage').style.display= "block";
			}
			else
			{
				document.getElementById('menu').style.position = "relative";
				document.getElementById('fixPage').style.display = "none";
			}
		}
	</script>
</head>
<body>
	<?php
function getDirs($dir)
{
	$dirfiles1 = scandir($dir);
	
	//Add files and details to new array
	foreach($dirfiles1 as $dirfile)
	{
		if($dirfile != '.' && $dirfile != '..')
		{
			$dir1 = explode('./', $dir);
			if ($dir1[1] != null)
			{
				$dir1 = $dir1[1].'/';
			}
			else
			{
				$dir1 = '';
			}
			if(is_dir($dir.'/'.$dirfile))
			{
				foreach(new RecursiveIteratorIterator(new RecursiveDirectoryIterator($dir1.$dirfile)) as $file)
				{
					$filesize += $file->getSize();
				}
			}
			else
			{
				$filesize = filesize($dir1.$dirfile);
			}
			$filesize = round($filesize,2);
			$filedate = date('Y-m-d H:i:s', filectime($dir1.$dirfile));
			array_push($GLOBALS['dirfiles'], array($dirfile, $dir, $filesize, $filedate));
			if(is_dir($dir.'/'.$dirfile)) 
			{
				getDirs($dir.'/'.$dirfile);
			}
		}
	}
}
 

function sortDirs()
{
	// Define the custom sort function
	function custom_sort($a,$b) 
	{
		switch($_GET['s'])
		{
			case 1:
				return $a[0]>$b[0];
				break;
			case 2:
				return $a[0]<$b[0];
				break;
			case 3:
				
				$a[0] = pathinfo($a[0], PATHINFO_EXTENSION);
				$b[0] = pathinfo($b[0], PATHINFO_EXTENSION);
				
				return $a[0]>$b[0];
				break;
			case 4:
				$a[0] = pathinfo($a[0], PATHINFO_EXTENSION);
				$b[0] = pathinfo($b[0], PATHINFO_EXTENSION);
				
				return $a[0]<$b[0];
				break;
			case 5:
				return $a[2]>$b[2];
				break;
			case 6:
				return $a[2]<$b[2];
				break;
			case 7:
				return $a[3]>$b[3];
				break;
			case 8:
				return $a[3]<$b[3];
				break;
			default:
				return $a[0]>$b[0];
		}
	}
	usort($GLOBALS['dirfiles'], "custom_sort");
}

function printDirs($dir, $isSub)
{
	//$dirfiles = scandir($dir);
	
	
	//Echo all files in directory
	foreach($GLOBALS['dirfiles'] as $dirfile)
	{
		if ($dir == $dirfile[1])
		{
			$dir = $dirfile[1];
			$dirfilename = $dirfile[0];
			$dirfiledate = $dirfile[3];
			$dirfilesize = $dirfile[2];
			if($dirfilename != '.' && $dirfilename != '..')
			{
				if(!is_dir($dir.'/'.$dirfilename))
				{
					$dir1 = explode('./', $dir);
					if ($dir1[1] != null)
					{
						$dir1 = $dir1[1].'/';
					}
					else
					{
						$dir1 = '';
					}
					$subPrefix = $isSub*40+10;
					$sizeunit = " B";
					if ($dirfilesize >= 1024)
					{
						$dirfilesize = $dirfilesize/1024;
						$sizeunit = " KB";
						if ($dirfilesize >= 1024)
						{
							$dirfilesize = $dirfilesize/1024;
							$sizeunit = " MB";
							if ($dirfilesize >= 1024)
							{
								$dirfilesize = $dirfilesize/1024;
								$sizeunit = " GB";
								if ($dirfilesize >= 1024)
								{
									$dirfilesize = $dirfilesize/1024;
									$sizeunit = " TB";
								}
							}
						}
					}
					$filesize = round($dirfilesize, 2).$sizeunit;
					$filedate = date('d-m-y H:i:s', filectime($dir1.$dirfilename));
					$filedetails = $filedate.'&nbsp;&nbsp;<span style="font-size: 24px; position: absolute; margin-left: -3px; margin-top: -8px;">|</span>&nbsp;&nbsp;'.$filesize;
					echo '<div id="a-'.$dir1.$dirfilename.'" class="file"><p style="margin-left: '.$subPrefix.'px;"><a href="'.$dir1.$dirfilename.'">/'.$dir1.$dirfilename.'</a></p></div>';
					echo '<p class="fileDetails">'.$filedetails.'</p></div>';
	
				}
			}
		}
	}
	
	//Echo all subdirectories
	foreach($GLOBALS['dirfiles'] as $dirfile)
	{
		if ($dir == $dirfile[1])
		{
			$dir = $dirfile[1];
			$dirfilename = $dirfile[0];
			$dirfiledate = $dirfile[3];
			$dirfilesize = $dirfile[2];
			if($dirfilename != '.' && $dirfilename != '..')
			{
				if(is_dir($dir.'/'.$dirfilename)) 
				{
					$dir1 = explode('./', $dir);
					if ($dir1[1] != null)
					{
						$dir1 = $dir1[1].'/';
					}
					else
					{
						$dir1 = '';
					}
					$subPrefix = $isSub*40+10;
					$sizeunit = " B";
					if ($dirfilesize >= 1024)
					{
						$dirfilesize = $dirfilesize/1024;
						$sizeunit = " KB";
						if ($dirfilesize >= 1024)
						{
							$dirfilesize = $dirfilesize/1024;
							$sizeunit = " MB";
							if ($dirfilesize >= 1024)
							{
								$dirfilesize = $dirfilesize/1024;
								$sizeunit = " GB";
								if ($dirfilesize >= 1024)
								{
									$dirfilesize = $dirfilesize/1024;
									$sizeunit = " TB";
								}
							}
						}
					}
					$filesize = round($dirfilesize, 2).$sizeunit;
					$filedate = date('d-m-y H:i:s', filectime($dir1.$dirfilename));
					$filedetails = $filedate.'&nbsp;&nbsp;<span style="font-size: 24px; position: absolute; margin-left: -3px; margin-top: -8px;">|</span>&nbsp;&nbsp;'.$filesize;
					echo '<div id="a-'.$dir1.$dirfilename.'" class="dir"><p style="margin-left: '.$subPrefix.'px;"><a href="'.$dir1.$dirfilename.'"><b>/'.$dir1.$dirfilename.'</b></a></p>';
					echo '<p class="fileDetails">'.$filedetails.'&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</p>';
					echo '<p class="anchorDir" onclick="location.href = \'#a-'.$dir1.$dirfilename.'\';">A</p>';
					if (isset($_GET['t']) && $_GET['t'] == 'c')
					{
						echo '<p class="toggleDir" id="toggle'.$dir1.$dirfilename.'"onclick="toggleDir(\''.$dir1.$dirfilename.'\');">&#60;</p>';
						echo '</div><span id="'.$dir1.$dirfilename.'" style="display: none;">';
					}
					else
					{
						echo '<p class="toggleDir" id="toggle'.$dir1.$dirfilename.'"onclick="toggleDir(\''.$dir1.$dirfilename.'\');">V</p>';
						echo '</div><span id="'.$dir1.$dirfilename.'" style="display: block;">';
					}
					printDirs($dir.'/'.$dirfilename, $isSub + 1);
					echo '</span>';
				}
			}
		}
	}
}

//Echo Header
echo '<div id="header"><h1>PHP Easy Directory Explorer</h1>';
echo '<p><b>Current relative path: '.$_SERVER['PHP_SELF'].'</b><br /><br /></p></div>';

//Add a sticky menu
echo '<div id="menu">';
echo '<p style="margin-left: 10px;"><a href="#">Go to top</a></p> ';
echo '<p class="menuSort" style="margin-left: 10px;">Sort&nbsp;by: &nbsp;&nbsp; ';
echo '<a href="'.$_SERVER['PHP_SELF'].'?s=1">Name&nbsp;(Asc)</a>&nbsp;&nbsp;/&nbsp;&nbsp;';
echo '<a href="'.$_SERVER['PHP_SELF'].'?s=2">Name&nbsp;(Desc)</a>';

echo ' &nbsp;&nbsp;&nbsp;/ &nbsp;&nbsp;&nbsp;';

echo '<a href="'.$_SERVER['PHP_SELF'].'?s=3">Type&nbsp;(Asc)</a>&nbsp;&nbsp;/&nbsp;&nbsp;';
echo '<a href="'.$_SERVER['PHP_SELF'].'?s=4">Type&nbsp;(Desc)</a>';

echo ' &nbsp;&nbsp;&nbsp;/ &nbsp;&nbsp;&nbsp;';

echo '<a href="'.$_SERVER['PHP_SELF'].'?s=5">Size&nbsp;(Asc)</a>&nbsp;&nbsp;/&nbsp;&nbsp;';
echo '<a href="'.$_SERVER['PHP_SELF'].'?s=6">Size&nbsp;(Desc)</a>';

echo ' &nbsp;&nbsp;&nbsp;/ &nbsp;&nbsp;&nbsp;';

echo '<a href="'.$_SERVER['PHP_SELF'].'?s=7">Date&nbsp;(Asc)</a>&nbsp;&nbsp;/&nbsp;&nbsp;';
echo '<a href="'.$_SERVER['PHP_SELF'].'?s=8">Date&nbsp;(Desc)</a>';

echo '</p></div>';

//Add a fix for sticky menu
echo '<div id="fixPage"><p style="margin-left: 10px;">Sort by: &nbsp;&nbsp;&nbsp; <p class="menuSort" style="margin-left: 10px;">Sort&nbsp;by: &nbsp;&nbsp;&nbsp; <a href="1">Name&nbsp;(Asc)&nbsp;&nbsp;/&nbsp;&nbsp;<a href="2">Name&nbsp;(Desc) &nbsp;&nbsp;&nbsp;/ &nbsp;&nbsp;&nbsp;<a href="3">Size&nbsp;(Asc)&nbsp;&nbsp;/&nbsp;&nbsp;<a href="4">Size&nbsp;(Desc) &nbsp;&nbsp;&nbsp;/ &nbsp;&nbsp;&nbsp;Date&nbsp;Modified&nbsp;(Asc)&nbsp;&nbsp;/&nbsp;&nbsp;Date&nbsp;Modified&nbsp;(Desc)</p></div></div>';

if(strpos($_SERVER['PHP_SELF'], 'dir.php') && file_exists('../dir.php') && file_exists('/dir.php'))
{
	echo '<div class="dir"><p style="margin-left: 10px"><a href="../dir.php"><b>Go Up</b></a> &nbsp;&nbsp; <a href="/dir.php"><b>Go To Root</b></a></p></div>';
}
else if(strpos($_SERVER['PHP_SELF'], 'dir.php') && file_exists('../dir.php'))
{
	echo '<div class="dir"><p style="margin-left: 10px"><a href="../dir.php"><b>Go Up</b></a> &nbsp;&nbsp; <a href="/"><b>Go To Root</b></a></p></div>';
}
else if(strpos($_SERVER['PHP_SELF'], 'dir.php') && file_exists('/dir.php'))
{
	echo '<div class="dir"><p style="margin-left: 10px"><a href="../"><b>Go Up</b></a> &nbsp;&nbsp; <a href="/dir.php"><b>Go To Root</b></a></p></div>';
}
else
{
	echo '<div class="dir"><p style="margin-left: 10px"><a href=".."><b>Go Up</b></a> &nbsp;&nbsp; <a href="/"><b>Go To Root</b></a></p></div>';
}

//Initiate the functions
$dirfiles = array();
getDirs('.');
sortDirs();
printDirs('.', 0);
	?>
	<br />
	<br />
	<p class="footerText">
		<a rel="license" href="http://creativecommons.org/licenses/by-nd/4.0/deed.en_US">
			<img alt="Creative Commons License" style="border-width:0;" src="http://i.creativecommons.org/l/by-nd/4.0/80x15.png" />
		</a>
		&nbsp;&nbsp;
		<span xmlns:dct="http://purl.org/dc/terms/" property="dct:title">
			PHP Easy Directory Explorer
		</span> 
		by 
		<a xmlns:cc="http://creativecommons.org/ns#" href="https://twitter.com/__Caspar__" property="cc:attributionName" rel="cc:attributionURL">
			Caspar Neervoort
		</a> 
		is licensed under a 
		<a rel="license" href="http://creativecommons.org/licenses/by-nd/4.0/deed.en_US">
			Creative Commons Attribution-NoDerivatives 4.0 International License
		</a>.
	</p>
</body>
</html>

EDEupdatehelper.php
<!--
PHP Easy File Explorer
Author: Caspar Neervoort
Creative Commons Attribution-NoDerivatives 4.0 International License.

REQUIRES: PHP 5.3 or higher.
You may use this anywhere and in any way you want, as long as you don`t alter the license information.
If you want to adapt this script, please contact me for permission and further information.

INSTRUCTIONS:
Do not use this file on it`s own, it updates the installer to the latest version.
-->

<html>
<head>
	<META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
</head>
<body>
	<?php
	if($installKey == $_GET['upd'])
	{
		try
		{
			ob_start();
			include $_SERVER['DOCUMENT_ROOT'].'/PHPEasyDirectoryExplorer.php';
			$source = file_get_contents('http://www.armorwatcher.com/PHPEasyDirectoryExplorer/Sources/PHPEasyDirectoryExplorer.txt');
			$source = str_replace('#k#e#y#', $installKey, $source);
			file_put_contents('./PHPEasyDirectoryExplorer.php', $source);
			ob_clean();
			echo 'DONE';
			echo '<script>location = \'./PHPEasyDirectoryExplorer.php\'</script>';
		}
		catch(Exception $e)
		{
			echo 'ERROR, please contact the author.<br />'.$e;
		}
	}
	else
	{
		echo '<script>alert(\'That key was not valid!\nCould not update.\');</script>';
		echo '<script>location = \'./PHPEasyDirectoryExplorer.php\'</script>';
	}
	?>
</body>
</html>

PHPEasyDirectoryExplorer.php
<!--
PHP Easy File Explorer
Author: Caspar Neervoort
Creative Commons Attribution-NoDerivatives 4.0 International License.

REQUIRES: PHP 5.3 or higher.
You may use this anywhere and in any way you want, as long as you don`t alter the license information.
If you want to adapt this script, please contact me for permission and further information.

INSTRUCTIONS:
To get the intall key of $installKey below this line in the source of this file.
<?php $installKey = '00100'; //Change 00100 to a custom key once as soon as you can! ?>
Remember to change this key once you`ve installed PHP Easy Directory Explorer!
-->

<html>
<head>
	<title>Install Easy Directory Explorer</title>
	<META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
	<style>
		body
		{
			margin: 0;
		}
		h1
		{
			font-family: Arial, Verdana, sans-serif;
		}
		p
		{
			font-family: Arial, Verdana, sans-serif;
			font-size: 12px;
			margin-top: 10px;
			margin-bottom: 12px;
		}
		legend
		{
			font-family: Arial, Verdana, sans-serif;
			font-size: 12px;
			margin-top: 10px;
			margin-bottom: 12px;
		}
		#header
		{
			z-index: 500;
			padding-left: 5px; 
			padding-top: 5px; 
		}
		.legalstuff
		{
			color: #999;
		}
		#changelog
		{
			display: none; 
		}
		.footerText
		{
			color: #999;
		}
		ul
		{
			margin: 0; 
			padding: 0;
			list-style-type: none;
		}
		
		ul li a
		{
			text-decoration: none;
			border-top: #ccc solid 1px;
			color: #315ADD; 
			padding: 10px 11px;
			background: #ddd;
			display:block;
		}
		
		ul li a:hover
		{
			background-color: #ededed;
		}
	</style>
	<script>
		function pingInfo(link, callback) 
		{
		    var xhr = new XMLHttpRequest();
		    xhr.open("GET", link, true);
		    xhr.onreadystatechange = function() 
		    {
		        if (xhr.readyState === 4) 
		        {
		            callback(xhr.responseText);
		        }
		    };
		    xhr.send(null);
		}
		
		function installEDE()
		{
			key = document.getElementById("key").value
			if(document.getElementById("sub").checked) {sub = '1';} else {sub = '0';}
			if(document.getElementById("ind").checked) {ind = '1';} else {ind = '0';}
			if(document.getElementById("rmv").checked) {rmv = '1';} else {rmv = '0';}
			if(!document.getElementById("lgl").checked) {alert('You need to agree to the legal terms in order to continue!'); return;}
			if(document.getElementById("key").value == '') {alert('You need to enter a key in order to continue!'); return;}
			
			location = 'PHPEasyDirectoryExplorer.php?key=' + key + '&sub=' + sub+ '&ind=' + ind + '&rmv=' + rmv;
		}
		
		function updateInstaller()
		{
			key = document.getElementById("key").value
			if(document.getElementById("key").value == '') {alert('You need to enter a key in order to continue!'); return;}
			location = 'PHPEasyDirectoryExplorer.php?upd=' + key;
		}
		
		function getChangelog()
		{
			pingInfo('http://www.armorwatcher.com/PHPEasyDirectoryExplorer/Sources/changelog.txt', function(text) 
			{
				document.getElementById('changelog').style.display = "block";
				document.getElementById('changelog').innerHTML = "<p>Current version: <b>v1.08</b><br /><br />" + text + "</p>";
			});
		}
	</script>
</head>
<body>
	<div id="header">
	<h1>Install PHP Easy Directory Explorer</h1>
	<p>
	<?php
	function installDirs($dir, $indexName, $source) //Install in subdirectories
	{
		$dirfiles1 = scandir($dir);
		
		foreach($dirfiles1 as $dirfile)
		{	
			if(is_dir($dir.'/'.$dirfile)) 
			{
				if($dirfile != '.' && $dirfile != '..')
				{
					echo 'Installing into subdirectories as '.$indexName.'...<br />';
					file_put_contents($dir.'/'.$dirfile.'/'.$indexName, $source);
					echo 'Installation into '.$dir.'/'.$dirfile.' complete!<br /><br />';
					installDirs($dir.'/'.$dirfile, $indexName, $source);
				}
			}
		}
	}
	
	function removeDirs($dir, $indexName) //Remove from subdirectories
	{
		$dirfiles1 = scandir($dir);
		
		foreach($dirfiles1 as $dirfile)
		{	
			if(is_dir($dir.'/'.$dirfile)) 
			{
				if($dirfile != '.' && $dirfile != '..')
				{
					echo 'Removing '.$indexName.' from subdirectory...<br />';
					try
					{
						unlink($dir.'/'.$dirfile.'/'.$indexName);
						echo 'Removal from '.$dir.'/'.$dirfile.' complete!<br /><br />';
					}
					catch(Exception $e)
					{
						echo 'Removal from '.$dir.'/'.$dirfile.' failed!<br />';
						echo 'This is likely caused because the file does not exist.<br />';
						echo 'If this is not the case, and the file is still present, please contact the author with the following error:<br />';
						echo $e.'<br /><br />';
					}
					removeDirs($dir.'/'.$dirfile, $indexName, $source);
				}
			}
		}
	}
	
	if($_GET['ind'] == 1)
	{
		$indexName = 'index.php';
	}
	else if($_GET['ind'] == 0)
	{
		$indexName = 'dir.php';
	}
	
	if($installKey == $_GET['key'] && $_GET['rmv'] == 1)
	{
		echo '		Removing PHP Easy Directory Explorer for you...
				<br />
				Options: Including subdirectories, Removing '.$indexName.'.
				<br />
				<br />
			</p>
		</div>
		<fieldset class="formField">
			<legend>Progress...</legend>
			<p>';
			
			echo 'Removing '.$indexName.' from current directory...<br />';
			try
			{
				unlink($indexName);
				echo 'Removal from current directory complete!<br /><br />';
			}
			catch(Exception $e)
			{
				echo 'Removal from current directory failed!<br />';
				echo 'This is likely caused because the file does not exist.<br />';
				echo 'If this is not the case, and the file is still present, please contact the author with the following error:<br />';
				echo $e.'<br /><br />';
			}
			
			removeDirs('.', $indexName);
			echo 'Removal complete!<br />';
		echo '	</p>
		</fieldset>';
	}
	else if($installKey == $_GET['key'] && $_GET['sub'] == 1)
	{
		echo '		Installing PHP Easy Directory Explorer for you...
				<br />
				Options: Including subdirectories, Install as '.$indexName.'.
				<br />
				<br />
			</p>
		</div>
		<fieldset class="formField">
			<legend>Progress...</legend>
			<p>';
			echo 'Getting source file contents...<br />';
			$source = file_get_contents('http://www.armorwatcher.com/PHPEasyDirectoryExplorer/Sources/index.txt');
			echo 'Source file read into memory<br /><br />';
			
			echo 'Installing into current directory as '.$indexName.'...<br />';
			file_put_contents($indexName, $source);
			echo 'Installation into current directory complete!<br /><br />';
			
			installDirs('.', $indexName, $source);
			echo 'Installation complete!<br />';
		echo '	</p>
		</fieldset>';
	}
	else if($installKey == $_GET['key'] && $_GET['sub'] == 0)
	{
		echo '		Installing PHP Easy Directory Explorer for you...
				<br />
				Options: No subdirectories, Install as '.$indexName.'.
				<br />
				<br />
			</p>
		</div>
		<fieldset class="formField">
			<legend>Progress...</legend>
			<p>';
			echo 'Getting source file contents...<br />';
			$source = file_get_contents('http://www.armorwatcher.com/PHPEasyDirectoryExplorer/Sources/index.txt');
			echo 'Source file read into memory<br /><br />';
			
			echo 'Installing into current directory as '.$indexName.'...<br />';
			file_put_contents($indexName, $source);
			echo 'Installation complete!<br />';
		echo '	</p>
		</fieldset>';
	}
	else if($installKey == $_GET['upd'])
	{
		$source = file_get_contents('http://www.armorwatcher.com/PHPEasyDirectoryExplorer/Sources/EDEupdatehelper.txt');
		file_put_contents('./EDEupdatehelper.php', $source);
		echo '<script>location = \'./EDEupdatehelper.php\';</script>';
	}
	else if(isset($_GET['key']) && $installKey != $_GET['key'])
	{
		echo '<script>alert(\'That key was not valid!\nCould not install.\');</script>';
		echo '<script>location = \'./PHPEasyDirectoryExplorer.php\'</script>';
	}
	else if(isset($_GET['upd']) && $installKey != $_GET['upd'])
	{
		$source = file_get_contents('http://www.armorwatcher.com/PHPEasyDirectoryExplorer/Sources/EDEupdatehelper.txt');
		file_put_contents('./EDEupdatehelper.php', $source);
		echo '<script>alert(\'That key was not valid!\nCould not update.\');</script>';
		echo '<script>location = \'./PHPEasyDirectoryExplorer.php\'</script>';
	}
	else
	{
		echo '		Enter your custom key to install or update PHP Easy Directory Explorer in the current and all subdirectories.
				<br />
				You can find your key by following the instructions in the commented section at the beginning of this document.
				<br />
				<br />
				<b>REMEMBER TO CHANGE YOUR KEY ONCE YOU\'VE FINISHED INSTALLING!</b>
				<br />
				Failure to do so may result in <u>a loss of files</u> if people figure out you\'re using the default key, which is 00100.
				<br />
				<br />
				<b>Current relative path: '.$_SERVER['PHP_SELF'].'</b>
				<br />
				<br />
			</p>
		</div>
		<fieldset class="formField">
			<legend>Install / Update</legend>
			<p>
				Key:
				<br />
				<input id="key" type="text" required="">
				<br />
				<br />
				<input id="sub" type="checkbox" value="1">
				<label for="sub"> Include subdirectories?</label>
				<br />
				<br />
				<input id="ind" type="checkbox" value="1">
				<label for="index"> Install as index.php? (default is dir.php. This WILL overwrite any current files of the same name!)</label>
				<br />
				<br />
				<input id="rmv" type="checkbox" value="1">
				<label for="rmv"> Uninstall? (make sure the above setting was the same as during install!) </label>
				<br />
				<br />
				<input id="lgl" type="checkbox" value="1" required="">
				I understand and agree with the terms below:
				<span class="legalstuff">
				<br /> 
				You understand and agree that the author will not be held responsible for what you, the user, do with this product.
				<br /> 
				In the event of any lost data or other inconveniences, the user is at fault for not reading or following the instructions properly.
				<br /> 
				Using this product is at your own risk.
				</span>
			</p>
			<button onclick="installEDE();">Install/Update</button>
		</fieldset>';
	}

	?>
	
	<br />
	<fieldset>
		<legend>Other functions:</legend>
		<button onclick="updateInstaller();">Update Installer</button>
		<button onclick="getChangelog();">Get Changelog</button>
		<button onclick="window.location = 'index.php';">Go to index.php</button>
		<button onclick="window.location = 'dir.php';">Go to dir.php</button>
	</fieldset>
	<fieldset id="changelog">
	</fieldset>
	<br />
	<br />
	<p class="footerText">
		<a rel="license" href="http://creativecommons.org/licenses/by-nd/4.0/deed.en_US">
			<img alt="Creative Commons License" style="border-width:0;" src="http://i.creativecommons.org/l/by-nd/4.0/80x15.png" />
		</a>
		&nbsp;&nbsp;
		<span xmlns:dct="http://purl.org/dc/terms/" property="dct:title">
			PHP Easy Directory Explorer
		</span> 
		by 
		<a xmlns:cc="http://creativecommons.org/ns#" href="https://twitter.com/__Caspar__" property="cc:attributionName" rel="cc:attributionURL">
			Caspar Neervoort
		</a> 
		is licensed under a 
		<a rel="license" href="http://creativecommons.org/licenses/by-nd/4.0/deed.en_US">
			Creative Commons Attribution-NoDerivatives 4.0 International License
		</a>.
	</p>
</body>
</html>